### PR TITLE
Support local backends

### PR DESCRIPTION
**Pull request by jgoizueta**
I'd like to merge this because it's becoming a pain to maintain these change!

This is to be able to use a local backend (which requires specific ports in the Maps & SQL API URLs).

A local environment can be used by using a `serverURL` like this: `http://{user}.localhost.lan{local}`.
The `{local}` placeholder is to allow insertion of the ports, shouldn't be present in non-local URLs.

I've also completed support of (optional) API keys (which was lacking in some calls to the SQL API

### CODE REVIEW

**Comment by jgoizueta** [comment]
cc @Jesus89 @IagoLast don't mean to bother you with boring stuff, but feel free to CR too

**Comment by Jesus89** [comment]
Do we want this changes for the `alpha.1` tag?

**Comment by jgoizueta** [comment]
Sorry for the messy commits guys 😅 I'm not sure what I did, but things look OK now, the only relevant commits are 013e6ef and a7477ea

**Comment by jgoizueta** [comment]
This shouldn't be needed for alpha.1 since local envs or API keys are not going to be used in that release.

But I'll be very happy if this is merged before new conflicts arise 😅

**Comment by Jesus89** [comment]
OK. It looks good to me, btw

File: src/client/windshaft.js [hunk 1]
```diff
@@ -552,12 +554,13 @@ export default class Windshaft {
     }
 
     async getDatesTypes(names, query, conf) {
+        console.log('GETDATAT',conf);
```
Comment (IagoLast) [inline src/client/windshaft.js]: do we want this? :P

File: src/client/windshaft.js [hunk 2]
```diff
@@ -611,16 +614,35 @@ function getAggFN(name) {
...
+    return endpoint(conf, `${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt`);
+}
+
+function getSQL(query, conf) {
+    let url = `${conf.sqlServerURL}/api/v2/sql?q=` + encodeURIComponent(query);
+    url = authURL(url, conf);
+    return fetch(url);
+}
+
+function authURL(url, conf) {
```
Comment (IagoLast) [inline src/client/windshaft.js]: `addApiKey` would be a better name IMO

File: src/client/windshaft.js (hunk 1 above)
Comment (jgoizueta) [reply src/client/windshaft.js]: No!

File: src/client/windshaft.js [hunk 3]
```diff
@@ -611,16 +614,35 @@ function getAggFN(name) {
...
-        return `${urlTemplates.url}/${conf.username}/api/v1/map/${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt?api_key=${conf.apiKey}`;
+        return authURL(`${urlTemplates.url}/${conf.username}/api/v1/map/${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt`, conf);
+    }
+    return endpoint(conf, `${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt`);
+}
+
+function getSQL(query, conf) {
+    let url = `${conf.sqlServerURL}/api/v2/sql?q=` + encodeURIComponent(query);
+    url = authURL(url, conf);
+    return fetch(url);
```
Comment (IagoLast) [inline src/client/windshaft.js]: maybe this function could return the `parsed json` so in the places where its used we get rid of the 

```javascript
const response = await getSql(q, c);
const sql = await response.json();
```

File: src/api/source/base-windshaft.js [hunk 4]
```diff
@@ -39,8 +39,11 @@ export default class BaseWindshaft extends Base {
     _generateURL(auth, config) {
```
Comment (IagoLast) [inline src/api/source/base-windshaft.js]: maybe we should rename url to urlS since this variable now contains an object with multiple urls inside

**Pull request merged**